skrf/io/touchstone.py

```diff
@@ -105,7 +117,34 @@ def __init__(self, file: typing.Union[str, typing.TextIO]):
         self.port_names = None
 
         self.comment_variables = None
-        self.load_file(fid)
+        
+        # open the file depending on encoding
+        # Guessing the encoding by trial-and-error, unless specified encoding
```

Comment:
If a encoding is passed and causes an Exception, there still an trial-and error approach with the two latest encodings, however the default encoding is skipped.
Why not adding all encodings, plus the custom one into a list and try all of them by default.

```python
encodings = ['latin1', 'ISO-8859-1', 'utf-8-sig']
encodings.insert(0, encoding) # Default encoding is None anyway

fid = None
for enc in encodings:
    try:
        fid = get_fid(file, encoding=encoding)
        break
    except ValueError:
        pass

if not fid:
    raise ValueError(f'Something went wrong by opening the file opening: {e}')

self.filename = fid.name
self.load_file(fid)
```

Reply:
Maybe it's even easier to try enough encodings by default and omit the additional parameter

Reply:
While it's more elegant, it would still fail if one tries to open a Windows-like UTF-8 file (for example). Because `get_fid` would returns a fid when reading the file as 'latin-1', but `load_file` would fail after due to the BOM of the file. And if it's not `load_file`, then it would be `get_gamma_z0_from_fid` to fail.